documentation/modules/cruise-control/con-cruise-control-overview.adoc

```diff
@@ -2,54 +2,508 @@
+
+* `Kafka` resource for main, hard, and default goals.
+** Main goals: `Kafka.spec.cruiseControl.config.goals`
+** Hard goals: `Kafka.spec.cruiseControl.config.hard.goals`
+** Default goals: `Kafka.spec.cruiseControl.config.default.goals`
+* `KafkaRebalance` resource for proposal-specific goals.
+** Proposal-specific goals: `KafkaRebalance.spec.goals`
+
+=== Main goals
+
+Main goals are predefined and available to all users in Cruise Control. 
+Goals not listed as main goals cannot be used in Cruise Control operations.
+Some main goals are preset as hard goals.
+
+To simplify configuration, use the inherited main goals unless you need to exclude specific goals from `KafkaRebalance` resources. 
+You can adjust the priority order in the default optimization goals configuration.
+
+Configure main goals in `Kafka.spec.cruiseControl.config.goals`:
+
+* To accept inherited main goals, omit the `goals` property.
+* To modify main goals, specify the goals in descending priority order in the `goals` property.
+
+=== Hard and soft goals
+
+Hard goals are mandatory and must be satisfied for optimization proposals to be generated. 
+Soft goals are best-effort objectives that Cruise Control tries to meet after all hard goals are satisfied.
+The classification of hard and soft goals is fixed in Cruise Control code and cannot be changed.
+
+Cruise Control first prioritizes satisfying hard goals, and then addresses soft goals in the order they are listed.
+A proposal meeting all hard goals is valid, even if it violates some soft goals. 
+
+For example, a soft goal might be to evenly distribute a topic's replicas. 
+Cruise Control will ignore this if it conflicts with hard goals.
+
+Configure hard goals in your Cruise Control deployment using `Kafka.spec.cruiseControl.config.hard.goals`:
+
+* To enforce all hard goals, omit the `hard.goals` property.
+* To specify hard goals, list them in `hard.goals`.
+* To exclude a hard goal, ensure it's not in either `default.goals` or `hard.goals`.
+
+Increasing the number of configured hard goals will reduce the likelihood of Cruise Control generating optimization proposals.
+
+=== Default goals
+
+Cruise Control uses default goals to generate an optimization proposal. 
+You can override default goals by setting proposal-specific optimization goals in a `KafkaRebalance` resource.
+
+If `default.goals` is not specified in the Cruise Control deployment configuration, main goals are used as default goals. 
+The optimization proposal is then generated using these main goals.
+
+Configure default goals in `Kafka.spec.cruiseControl.config.default.goals`:
+
+* To use main goals as default, omit the `default.goals` property.
+* To modify default goals, specify a subset of main goals in the `default.goals` property.
+
+=== Proposal-specific goals
+
+Proposal-specific optimization goals narrow down the configured default goals for specific optimization proposals.
```

Comment:
Not necessarily "narrows down", the configured default goals list could be smaller than proposal-sepcific goals list.
```suggestion
Proposal-specific optimization goals allow creating optimization proposals based on a specific list of goals.
```